bigquery_etl/schema/__init__.py

```diff
@@ -46,13 +48,17 @@ def from_json(cls, json_schema):
     @classmethod
     def for_table(cls, project, dataset, table, partitioned_by=None):
         """Get the schema for a BigQuery table."""
-        query = f"SELECT * FROM {project}.{dataset}.{table}"
+        query = f"SELECT * FROM `{project}.{dataset}.{table}`"
 
         if partitioned_by:
             query += f" WHERE DATE({partitioned_by}) = DATE('2020-01-01')"
 
         # write query to temporary file so it can get dry run
-        tmp = NamedTemporaryFile()
+        # ensure that directory structure corresponds to original structure,
+        # otherwise dry run might fail when extracting the dataset from the path
```

Comment:
I'm referring to https://github.com/mozilla/bigquery-etl/blob/8a55e8f429a76a309eb0568c2ca25adb918cf64b/bigquery_etl/dryrun.py#L262 here. By default when creating a temporary file in the Circle CI step, it will write the file to `tmp/some-random-name` which causes a failure in `Dryrun`.

Reply:
I've written shim code like this before, but I think you can pass content directly in the `DryRun` constructor rather than writing to a temp file, in which case we don't need to worry about directory structure. But I may be missing a detail here.

Reply:
Indeed, you are right. No need for temporary files.